mgmtsystem/azure.py

```diff
@@ -440,6 +440,42 @@ def remove_blob_image(self, vm_vhd, container=None):
                        storage_key=self.storage_key,
                        container=container), True)
 
+    def remove_unused_nics(self, nic_template):
+        """
+        Used for clean_up jobs to remove NIC that are not attached to any test VM
+        """
+        self.logger.info(" Removing NICs ")
```

Comment:
**REQUIRED** Let's put the nic_template into the removal too
**REQUIRED** I also think the function should be renamed to show that it removes based on a search/ie....it could remove all if you put *